km_api/know_me/tests/models/test_emergency_contact_model.py

```diff
@@ -0,0 +1,83 @@
+from know_me import models
+from django.core.exceptions import ValidationError
+from rest_framework.reverse import reverse
+import pytest
+
+
+def test_create(km_user_factory):
+    """
+    Test creating an emergency contact.
+    """
+    models.EmergencyContact.objects.create(
+        km_user=km_user_factory(),
+        name='First Last',
+        relation='Parent',
+        phone_number='19199199199',
+        alt_phone_number='12345678909',
+        email='hi@gmail.com')
+
+
+def test_get_absolute_url(emergency_contact_factory):
+    """
+    This method should return the URL of the emergency contact's
+    detail view.
+    """
+    ec = emergency_contact_factory()
+    expected = reverse(
+        'know-me:emergency-contact-detail',
+        kwargs={'pk': ec.pk}
+    )
+
+    assert ec.get_absolute_url() == expected
+
+
+def test_has_object_read_permission_owner(api_rf, emergency_contact_factory):
+    """
+    Users should have read access on their own emergency contacts.
+    """
+    ec = emergency_contact_factory()
+
+    api_rf.user = ec.km_user.user
+    request = api_rf.get('/')
+
+    assert ec.km_user.has_object_read_permission(request)
```

Comment:
This is testing that we have read permissions on the associated `km_user`, not on the emergency contact.